packages/modules/src/SessionKeyManagerModule.ts

```diff
@@ -1,139 +1,203 @@
     return setMerkleRootData
   }
 
-  async signUserOp(userOp: UserOperation): Promise<string> {
+  /**
+   * This method is used to sign the user operation using the session signer
+   * @param userOp The user operation to be signed
+   * @param sessionSigner The signer to be used to sign the user operation
+   * @returns The signature of the user operation
+   */
+  async signUserOp(userOp: UserOperation, sessionSigner: Signer): Promise<string> {
+    if (!sessionSigner) {
+      throw new Error('Session signer is not provided.')
+    }
+    // Use the sessionSigner to sign the user operation
     const userOpHash = getUserOpHash(userOp, this.entryPointAddress, this.chainId)
-    //
-    const signature = await this.sessionSigner.signMessage(arrayify(userOpHash))
-    // add validator module address to the signature
+    const signature = await sessionSigner.signMessage(arrayify(userOpHash))
+
+    const sessionSignerData = await this.sessionStorageClient.getSessionData({
+      sessionPublicKey: await sessionSigner.getAddress()
+    })
 
-    // Review // Should be done on account side
+    // Generate the padded signature with (validUntil,validAfter,sessionVerificationModuleAddress,validationData,merkleProof,signature)
+    const paddedSignature = defaultAbiCoder.encode(
+      ['uint48', 'uint48', 'address', 'bytes', 'bytes32[]', 'bytes'],
+      [
+        sessionSignerData.validUntil,
+        sessionSignerData.validAfter,
+        sessionSignerData.sessionValidationModule,
+        sessionSignerData.sessionKeyData,
+        this.merkleTree.getHexRoot(),
+        signature
+      ]
+    )
+
+    // Generate the encoded data with paddedSignature and sessionKeyManagerModuleAddress
     const signatureWithModuleAddress = ethers.utils.defaultAbiCoder.encode(
       ['bytes', 'address'],
-      [signature, this.getAddress()]
+      [paddedSignature, this.getAddress()]
     )
-    userOp.signature = signatureWithModuleAddress
-    // TODO: return userOp or signatureWithModuleAddress?
     return signatureWithModuleAddress
   }
 
-  async signMessage(message: Bytes | string): Promise<string> {
-    return await this.sessionSigner.signMessage(message)
+  /**
+   * Update the session data pending state to active
+   * @param param The search param to find the session data
+   * @param status The status to be updated
+   * @returns
+   */
+  async updateSessionStatus(param: SessionSearchParam, status: SessionStatus) {
+    this.sessionStorageClient.updateSessionStatus(param, status)
   }
 
-  async getMerkleProof(): Promise<string> {
-    // TODO: use nodeclient / local storage to get merkle proof
-
-    // const merkleProofData = await this.nodeClient.getMerkleProof(
-    //   this.sessionKeyModule,
-    //   this.sessionKey.getAddress()
-    // )
-    // console.log(merkleProofData)
+  /**
+   * @remarks This method is used to clear all the pending sessions
+   * @returns
+   */
+  async clearPendingSessions() {
+    this.clearPendingSessions()
```

Comment:
This should be `this.sessionStorageClient.clearPendingStatus()` otherwise this'll go in recursion forever